sap-spring-properties/src/main/java/com/sap/cloud/environment/servicebinding/environment/SapServiceBindingEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,37 @@
+package com.sap.cloud.environment.servicebinding.environment;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.context.properties.bind.BindResult;
+import org.springframework.boot.context.properties.bind.Bindable;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.annotation.Order;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import com.sap.cloud.environment.servicebinding.SapServiceBindingsProperties;
+
+/* An {@link EnvironmentPostProcessor} that binds service binding properties to {@link SapServiceBindingsProperties}. */
+@Order
+public class SapServiceBindingEnvironmentPostProcessor implements EnvironmentPostProcessor
+{
+    private static final Logger log = LoggerFactory.getLogger(SapServiceBindingEnvironmentPostProcessor.class);
+
+    @Override
+    public void postProcessEnvironment( ConfigurableEnvironment environment, SpringApplication application )
+    {
+        Bindable<SapServiceBindingsProperties> bindable = Bindable.of(SapServiceBindingsProperties.class);
+
+        BindResult<?> bindResult = Binder.get(environment).bind("cds", bindable);
+        if( bindResult.isBound() ) {
+            SapServiceBindingsPropertiesAccessor
+                .setServiceBindingsProperties(((SapServiceBindingsProperties) bindResult.get()).getServiceBindings());
+        } else if( bindable.getValue() != null ) {
```

Comment:
When would `bindable.getValue()` be set, but `bindResult.isBound()` returned false?

Reply:
That`s a good question I stumbled across too, when I wrote the unit tests. I just copied the code from CAP: com.sap.cds.framework.spring.config.runtime.SpringPropertiesProvider
I thought there must be a good reason for it that I don't know...

Reply:
In that class we also use `Bindable.ofInstance` and not only `Bindable.of`, therefore we need the additional `else if`. As you only use `Bindable.of` here, checking `bindResult.isBound()` is enough,

Reply:
oh, details :-) thanks for the explanation